token-minter/sources/modules/signed_transaction.move

```diff
@@ -0,0 +1,97 @@
+module minter::signed_transaction {
+    use std::ed25519;
+    use std::error;
+    use std::signer;
+    use std::string::String;
+    use aptos_framework::object;
+    use aptos_framework::object::{ConstructorRef, Object};
+
+    /// The proof challenge is invalid.
+    const EINVALID_CHALLENGE: u64 = 1;
+    /// The proof data does not exist.
+    const EPROOF_DATA_DOES_NOT_EXIST: u64 = 2;
+    /// The proof data already exists.
+    const EPROOF_DATA_ALREADY_EXISTS: u64 = 3;
+    /// The signer is not the owner of the object.
+    const ENOT_OBJECT_OWNER: u64 = 4;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct ProofData has key {
+        authority_public_key: ed25519::UnvalidatedPublicKey,
+    }
+
+    struct ProofChallenge has copy, drop {
+        data: vector<u8>,
+    }
+
+    /// Initializes the collection object's constructor ref with proof data.
+    public fun init(constructor_ref: &ConstructorRef, authority_public_key: vector<u8>): Object<ProofData> {
+        let object_signer = object::generate_signer(constructor_ref);
+        extend(&object_signer, authority_public_key)
+    }
+
+    /// Extends the collection object with proof data via the collection signer.
+    public fun extend(object_signer: &signer, authority_public_key: vector<u8>): Object<ProofData> {
+        let object_signer_addr = signer::address_of(object_signer);
+        assert!(!proof_data_exists(object_signer_addr), EPROOF_DATA_ALREADY_EXISTS);
```

Comment:
should we allow an object contain multiple ProofData? so that they could essentially be authorized as multiple admin accounts with multiple PKs?

Reply:
That sounds good, As long as any of the authority_public_keys match, then it can be deemed valid